anonymizer/base.py

```diff
@@ -71,9 +71,11 @@ def get_allowed_value(self, source, field):
 
     ### Public interace ##
     def uuid(self, field=None):
-        def source():
-            return str(uuid4())
-        return self.get_allowed_value(source, field)
+        retval = str(uuid4())
+        max_length = getattr(field, 'max_length', None)
+        if max_length is not None:
+            retval = retval[:max_length]
```

Comment:
a piece of trivia for you, the spec for uuid1 actually has alot of machine specific determinism, with lots of the trailing digits the same on the same machine

uuid4 on the other hand is completely pseudo-random